src/themes/teams/components/Divider/dividerStyles.ts

```diff
@@ -4,7 +4,7 @@ import {
   ICSSInJSStyle,
   ICSSPseudoElementStyle,
 } from '../../../../../types/theme'
-import { IDividerProps } from '../../../../components/Divider/Divider'
+import { IDividerPropsWithDefaults } from '../../../../components/Divider/Divider'
```

Comment:
What is the benefit of having additional interface with defaults if we have the IDividerProps interface? It seems confusing to me to write the `WithDefaults` every time I want to use the interface for the component in some place.

Reply:
in that case you mau use regular `IDividerProps` interface, this one is exposed as well. The problem comes where there is need to consume some property from `IDividerProps` for providing styles:

```jsx
summary: (props) => ({
   ... 
   content: 'Provided content has ${props.content.length} characters.'
  ...
})
```

here the problem is that, for the sake of making possible to omit `content` property while using component by the client, you may want to make this property being optional for props interface

```jsx
interface IComponentProps = {
   ...
   content?: string
}
```

At the same time, it could be easily the case that some default value is provided as part of `defaultProps` for it. In that case you might want to utilize this knowledge while applying styles - this is the reason for `IComponentPropsWithDefaults` to exist. If you don't need to know that (or rely on the fact that this particular property is initialized as default) in your styles, just use regular `IComponentProps` interface then, as usual

Reply:
do we want to have themes depend on components? should we rather extract all interfaces somewhere else? @mnajdova 
@kuzhelov

Reply:
agreed, thx for detailed answer @kuzhelov